### PR TITLE
Add CODEOWNERS usage details

### LINKED ISSUE #7783: Add CODEOWNERS usage details
### What article on docs.github.com is affected?

https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners#syntax-exceptions

### What part(s) of the article would you like to see updated?

Add usage details about:
- Excluding directories (using @ghost)
- Changing files under an author's ownership
- Appointing more than one CODEOWNER

### PR DESCRIPTION
**Pull request by deadletterq**
### Why:

Closes https://github.com/github/docs/issues/7783

### What's being changed:

- Usage of @ghost to exclude directories
- Changing files under an author's ownership
- Appointing more than one CODEOWNER

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
@sidletsky Thanks so much for opening an issue and a PR! I'll get this triaged for review ⚡

**Comment by felicitymay** [comment]
Thanks for this pull request ✨ 

I've asked a few questions about these changes on https://github.com/github/docs/issues/7783 and will review the pull request once we are clear on what changes to make to this article.

File: content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md [hunk 1]
```diff
@@ -84,13 +84,44 @@ apps/ @octocat
...
+/apps/github 
 ```
 ### Syntax exceptions
 There are some syntax rules for gitignore files that do not work in CODEOWNERS files:
 - Escaping a pattern starting with `#` using `\` so it is treated as a pattern and not a comment
 - Using `!` to negate a pattern
 - Using `[ ]` to define a character range
 
+## CODEOWNERS and branch protection
+It is possible to add a branch protection rule to ensure that changed code is reviewed by the owners of the changed files, see "[About protected branches](/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)."
```
Comment (felicitymay) [inline content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md:101]: Some minor updates for consistency with our style guide:

```suggestion
Repository owners can add branch protection rules to ensure that changed code is reviewed by the owners of the changed files. For more information, see "[About protected branches](/github/administering-a-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)." 
```

File: content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md [hunk 2]
```diff
@@ -84,13 +84,44 @@ apps/ @octocat
...
+
+### Example of a CODEOWNERS file
+```
+# In this example, any change inside `/apps` directory
+# will require approval from @doctocat.
+/apps/ @doctocat
+
+# In this example, any change inside `/apps` directory
+# will require approval from @doctocat or @octocat.
+/apps/ @doctocat @octocat
```
Comment (felicitymay) [inline content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md:112]:
```suggestion
# In this example, any change inside the `/apps` directory
# will require approval from @doctocat.
/apps/ @doctocat

# In this example, any change inside the `/apps` directory
# will require approval from @doctocat or @octocat.
/apps/ @doctocat @octocat
```

File: content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md [hunk 3]
```diff
@@ -84,13 +84,44 @@ apps/ @octocat
...
+
+# In this example, assuming @doctocat is a member 
+# of @Github/content-team and @doctocat introduces a 
+# change inside `/apps` directory, the approval 
+# can be done by anyone else, including the users 
+# that are not in @Github/content-team. 
+# @Github/content-team will be still appointed
+# to the review, but it's member's approval
+# is not required. 
+/apps/ @Github/content-team
```
Comment (felicitymay) [inline content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md:122]: I've re-written this a bit to try and simplify it. 

```suggestion
# In this example, any change inside the `/apps` directory
# will require approval from a member of the @example-org/content team.
# If a member of @example-org/content opens a pull request 
# with a change inside the `/apps` directory, their approval is implicit.
# The team is still added as a reviewer but not a required reviewer.
# Anyone can approve the changes.
/apps/ @example-org/content-team
```

**Review by felicitymay — COMMENTED** [review]
Hi @sidletsky -  thank you for your updates to this PR 💖 

I've suggested some changes, but they're fairly small things. Once I've merged those changes, this should be ready to merge. Thank you so much for taking the time to raise this problem, to discuss the best way to make the behavior clear, and to improve the docs for other users. ✨

**Comment by felicitymay** [comment]
👋🏻 I'm not sure if you'll have seen these or not, but just in case. 

I have just deleted a couple of comments on your original changes that I wrote but didn't submit. It looked as if they reappeared as part of the review of the current content. Apologies for any confusion.

File: content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md [hunk 4]
```diff
@@ -84,13 +84,42 @@ apps/ @octocat
 # directory in the root of your repository and any of its
 # subdirectories.
 /docs/ @doctocat
+
+# In this example, @octocat owns any file in the `/apps` 
+# directory in the root of your repository except for the `/apps/github` 
+# repository, as its owners are left empty.
```
Comment (felicitymay) [inline content/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners.md:90]: Spotted a typo

```suggestion
# subdirectory, as its owners are left empty.
```

**Review by felicitymay — APPROVED** [review]

**Pull request merged**